core/server/common/src/main/java/alluxio/cli/validation/ClusterConfConsistencyValidationTask.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.cli.validation;
+
+import alluxio.Configuration;
+import alluxio.PropertyKey;
+
+import com.google.common.collect.Sets;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.Set;
+
+/**
+ * Task for validating system limit for current user.
+ */
+public final class ClusterConfConsistencyValidationTask extends AbstractValidationTask {
+
+  @Override
+  public TaskResult validate(Map<String, String> optionMap) throws InterruptedException {
+    Set<String> masters = new HashSet<>(Utils.readNodeList("masters"));
+    Set<String> workers = new HashSet<>(Utils.readNodeList("workers"));
+    Set<String> nodes = Sets.union(masters, workers);
+    Map<String, Properties> allProperties = new HashMap<>();
+    Set<String> propertyNames = new HashSet<>();
+    if (masters.isEmpty()) {
+      System.err.println("No master nodes specified in conf/masters file");
+      return TaskResult.SKIPPED;
+    }
+    if (workers.isEmpty()) {
+      System.err.println("No worker nodes specified in conf/workers file");
+      return TaskResult.SKIPPED;
+    }
+    TaskResult result = TaskResult.OK;
+    for (String node : nodes) {
+      Properties props = getNodeConf(node);
+      if (props == null) {
+        result = TaskResult.FAILED;
+        continue;
+      }
+      allProperties.put(node, props);
+      propertyNames.addAll(props.stringPropertyNames());
+    }
+    for (String propertyName : propertyNames) {
+      Set<String> targetNodes = nodes;
+      if (propertyName.startsWith("alluxio.user.")) {
```

Comment:
It's confusing to split the configuration checking logic between here and PropertyKey - could we move this logic to PropertyKey?

Reply:
The information of which component the property applies to is added to each property key.